### PR TITLE
SASS-1003 - Added CreateOrAmendEmploymentExpensesConnector (DES #1669)

### PR DESCRIPTION
**Pull request by Dula-J**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-CHANGE-ME](ADD URL HERE)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/models/EmploymentExpensesRequestModel.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+final case class ExpenseType(
+                        businessTravelCosts: BigDecimal,
+                        jobExpenses: BigDecimal,
+                        flatRateJobExpenses: BigDecimal,
+                        professionalSubscriptions: BigDecimal,
+                        hotelAndMealExpenses: BigDecimal,
+                        otherAndCapitalAllowances: BigDecimal,
+                        vehicleExpenses: BigDecimal,
+                        mileageAllowanceRelief: BigDecimal
```
Comment (BenCosfordHMRC) [inline app/models/EmploymentExpensesRequestModel.scala:31]: can reuse the existing ExpensesType case class here as all fields should be optional

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Could we look into reusing the existing expensesType as described by Ben. Could we also address the following warns :
Screenshot 2021-06-30 at 14 26 25

We are also missing test coverage on the CreateOrAmendEmploymentExpensesService. I think switching the controller over to an integration test should remedy this.
Screenshot 2021-06-30 at 15 01 30

Finally, could we address the following platform dependancies please.
Screenshot 2021-06-30 at 15 03 35

File: test/controllers/CreateOrAmendEmploymentExpensesControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/*
```
[the remark this answers is not in the document]
Comment (Dula-J) [reply test/controllers/CreateOrAmendEmploymentExpensesControllerSpec.scala:1]: We can add an integration test once we have an idea what the routes are

File: app/models/EmploymentExpensesRequestModel.scala (hunk 1 above)
Comment (Dula-J) [reply app/models/EmploymentExpensesRequestModel.scala:31]: Done

**Comment by Dula-J** [comment]
> Could we look into reusing the existing expensesType as described by Ben. Could we also address the following warns :
> Screenshot 2021-06-30 at 14 26 25
> 
> We are also missing test coverage on the CreateOrAmendEmploymentExpensesService. I think switching the controller over to an integration test should remedy this.
> Screenshot 2021-06-30 at 15 01 30
> 
> Finally, could we address the following platform dependancies please.
> Screenshot 2021-06-30 at 15 03 35

Resolved, and added a unit test for the service to increase the coverage in that area.

**Review by robert-webb-hmrc — APPROVED** [review]

**Pull request merged**